src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenAResolvePackageAssetsTask.cs

```diff
@@ -82,6 +84,41 @@ public void ItDoesNotHashDesignTimeBuild()
                 because: $"{nameof(task.DesignTimeBuild)} should not be included in hash.");
         }
 
+        [Fact]
+        public void It_does_not_error_on_duplicate_package_names()
+        {
+            string projectAssetsJsonPath = Path.GetTempFileName();
+            var assetsContent =  @"{
+  `version`: 3,
+  `targets`: {
+    `net5.0`: {
+      `Humanizer.Core/2.2.0`: {
+        `type`: `package`
+      },
+      `Humanizer.Core/2.2.1`: {
+        `type`: `package`
+      }
+    }
+  },
+  `project`: {
+    `version`: `1.0.0`,
+    `frameworks`: {
+      `net5.0`: {
+        `targetAlias`: `net5.0`
+      }
+    }
+  }
+}".Replace('`', '"');
+            File.WriteAllText(projectAssetsJsonPath, assetsContent);
+
+            var task = InitializeTask(out _);
+            task.ProjectAssetsFile = projectAssetsJsonPath;
+            task.TargetFramework = "net5.0";
+            new CacheWriter(task)
+                .Should()
+                .NotBeNull();
```

Comment:
I don't think we need the `.Should().NotBeNull()` calls here.  Calling a constructor will never return null, it will return an object or throw an exception.